v2/api/machinelearningservices/v1api20240401/structure.txt

```diff
@@ -3,6 +3,316 @@ github.com/Azure/azure-service-operator/v2/api/machinelearningservices/v1api2024
 ------------------------------------------------------------------------------------
 APIVersion: Enum (1 value)
 └── "2024-04-01"
+Registry: Resource
+├── Owner: resources/v1apiv20191001.ResourceGroup
+├── Spec: Object (15 properties)
+│   ├── AzureName: Validated<string> (1 rule)
+│   │   └── Rule 0: Pattern: "^[a-zA-Z0-9][a-zA-Z0-9\\-_]{2,32}$"
+│   ├── DiscoveryUrl: *string
```

Comment:
Is this another field that should be read-only but isn't?

Reply:
It can either set by the user or auto-generated if not